### PR TITLE
Added ability for login screen to clear password on failed attempt

### PR DESCRIPTION
**Pull request by chalettu**
@miq-bot add_label enhancement
@miq-bot add_label fine/no

### CODE REVIEW

**Comment by chalettu** [comment]
Fix for BZ https://bugzilla.redhat.com/show_bug.cgi?id=1089786

File: client/app/states/login/login.state.js [hunk 1]
```diff
@@ -74,8 +74,11 @@ function StateController (exception, $state, Text, RBAC, API_LOGIN, API_PASSWORD
         Session.destroy()
       }
     })
-    .catch(() => {
-      exception.catch('Login failed, possibly invalid credentials.')
+    .catch((response) => {
+      if (response.status === 401) {
+        vm.credentials.password = ''
+        Notifications.message('danger', '', __('Incorrect username or password.'), false)
```
Comment (AllenBW) [inline client/app/states/login/login.state.js]: If the response has a server message we should probably display that here

File: client/app/states/login/login.state.js [hunk 2]
```diff
@@ -74,8 +74,11 @@ function StateController (exception, $state, Text, RBAC, API_LOGIN, API_PASSWORD
         Session.destroy()
       }
     })
-    .catch(() => {
-      exception.catch('Login failed, possibly invalid credentials.')
+    .catch((response) => {
+      if (response.status === 401) {
+        vm.credentials.password = ''
```
Comment (AllenBW) [inline client/app/states/login/login.state.js]: should `vm.credentials.login` also be cleared out?

Comment (AllenBW) [reply client/app/states/login/login.state.js]: I realize ref was for only password to be blank, but if it was an unsuccessful login attempt :-/

Comment (chalettu) [reply client/app/states/login/login.state.js]: The RFE just requested password be cleared out.

File: client/app/states/login/login.state.js (hunk 1 above)
Comment (chalettu) [reply client/app/states/login/login.state.js]: We probably could but a couple of thoughts.  The message that comes back just says "Authentication failed".  At a minimum the message we display is a little more specific and it is also translated vs the raw message from the server.   Keep in mind, I didn't add this notification message, merely moved it from the factory up to the state that it pertains to.

Comment (AllenBW) [reply client/app/states/login/login.state.js]: And normally I wouldn't press on this matter, but my 👃 is in a bz right now where not showing the full error message issssssss no bueno https://bugzilla.redhat.com/show_bug.cgi?id=1498984

**Pull request merged**